src/util/api/users/index.ts

```diff
@@ -193,12 +194,17 @@ export function checkEVMSignPayload({
   return actualPayload;
 }
 
-export function userByEmailQuery(user, email) {
+export function userByEmailQuery(user, email, isEmailVerified = false) {
   return dbRef.where('email', '==', email).get().then((snapshot) => {
     snapshot.forEach((doc) => {
       const docUser = doc.id;
       if (user !== docUser) {
-        throw new ValidationError('EMAIL_ALREADY_USED');
+        const { evmWallet, likeWallet } = doc.data();
+        const payload = isEmailVerified ? {
+          evmWallet: maskString(evmWallet),
+          likeWallet: maskString(likeWallet, { start: 11 }),
+        } : null;
+        throw new ValidationError('EMAIL_ALREADY_USED', 400, payload);
```

Comment:
Maybe using an `if` block can show more explicitly about the role `isEmailVerified` is playing here
```suggestion
        let payload = null;
        if (isEmailVerified) {
          const { evmWallet, likeWallet } = doc.data();
          payload = {
            evmWallet: maskString(evmWallet),
            likeWallet: maskString(likeWallet, { start: 11 }),
          };
        }
        throw new ValidationError('EMAIL_ALREADY_USED', 400, payload);
```